packages/playwright/src/worker/testInfo.ts

```diff
@@ -400,23 +407,24 @@ export class TestInfoImpl implements TestInfo {
   // ------------ TestInfo methods ------------
 
   async attach(name: string, options: { path?: string, body?: string | Buffer, contentType?: string } = {}) {
-    this._attach(name, await normalizeAndSaveAttachment(this.outputPath(), name, options));
-  }
-
-  private _attach(name: string, attachment: TestInfo['attachments'][0]) {
     const step = this._addStep({
       title: `attach "${name}"`,
       category: 'attach',
     });
+    step.attachments.push(await normalizeAndSaveAttachment(this.outputPath(), name, options));
```

Comment:
This logic is making the code backwards. It seems like _attach should add the index of the last attachment to the current step instead.

Reply:
good call! made that change.